art/estimators/object_detection/tensorflow_faster_rcnn.py

```diff
@@ -0,0 +1,530 @@
+    """
+    This class implements a model-specific object detector using Faster-RCNN and TensorFlow.
+    """
+
+    @deprecated_keyword_arg("channel_index", end_version="1.5.0", replaced_by="channels_first")
+    def __init__(
+        self,
+        model: Optional["FasterRCNNMetaArch"] = None,
+        filename: Optional[str] = None,
+        url: Optional[str] = None,
+        images: Optional["Tensor"] = None,
+        sess: Optional["Session"] = None,
+        is_training: bool = False,
+        clip_values: Optional["CLIP_VALUES_TYPE"] = None,
+        channel_index=Deprecated,
+        channels_first: Optional[bool] = None,
+        preprocessing_defences: Union["Preprocessor", List["Preprocessor"], None] = None,
+        postprocessing_defences: Union["Postprocessor", List["Postprocessor"], None] = None,
+        preprocessing: "PREPROCESSING_TYPE" = (0, 1),
+        attack_losses: Tuple[str, ...] = (
+            'Loss/RPNLoss/localization_loss',
+            'Loss/RPNLoss/objectness_loss',
+            'Loss/BoxClassifierLoss/localization_loss',
+            'Loss/BoxClassifierLoss/classification_loss'
+        )
+    ):
+        """
+        Initialization of an instance TensorFlowFasterRCNN.
+
+        :param model: A TensorFlow Faster-RCNN model. The output that can be computed from the model includes a tuple
+                      of (predictions, losses, detections):
+
+                        - predictions: a dictionary holding "raw" prediction tensors.
+                        - losses: a dictionary mapping loss keys (`Loss/RPNLoss/localization_loss`,
+                                  `Loss/RPNLoss/objectness_loss`, `Loss/BoxClassifierLoss/localization_loss`,
+                                  `Loss/BoxClassifierLoss/classification_loss`) to scalar tensors representing
+                                  corresponding loss values.
+                        - detections: a dictionary containing final detection results.
+        :param filename: Name of the file.
+        :param url: Download URL.
+        :param images: Input samples of shape (nb_samples, height, width, nb_channels).
+        :param sess: Computation session.
+        :param is_training: A boolean indicating whether the training version of the computation graph should be
+                            constructed.
+        :param clip_values: Tuple of the form `(min, max)` of floats or `np.ndarray` representing the minimum and
+                            maximum values allowed for input image features. If floats are provided, these will be
+                            used as the range of all features. If arrays are provided, each value will be considered
+                            the bound for a feature, thus the shape of clip values needs to match the total number
+                            of features.
+        :param channel_index: Index of the axis in data containing the color channels or features.
```

Comment:
For new estimators we can ignore the channel_index argument and only use channels_first.
```suggestion
```

Reply:
Agree.